src/types/CodepointWidthDetector.cpp

```diff
@@ -9,392 +9,358 @@ namespace
-        UnicodeRange{ 0x1f250, 0x1f251, CodepointWidth::Wide },
-        UnicodeRange{ 0x1f260, 0x1f265, CodepointWidth::Wide },
-        UnicodeRange{ 0x1f300, 0x1f320, CodepointWidth::Wide },
-        UnicodeRange{ 0x1f32d, 0x1f335, CodepointWidth::Wide },
-        UnicodeRange{ 0x1f337, 0x1f37c, CodepointWidth::Wide },
-        UnicodeRange{ 0x1f37e, 0x1f393, CodepointWidth::Wide },
-        UnicodeRange{ 0x1f3a0, 0x1f3ca, CodepointWidth::Wide },
-        UnicodeRange{ 0x1f3cf, 0x1f3d3, CodepointWidth::Wide },
-        UnicodeRange{ 0x1f3e0, 0x1f3f0, CodepointWidth::Wide },
-        UnicodeRange{ 0x1f3f4, 0x1f3f4, CodepointWidth::Wide },
-        UnicodeRange{ 0x1f3f8, 0x1f43e, CodepointWidth::Wide },
-        UnicodeRange{ 0x1f440, 0x1f440, CodepointWidth::Wide },
-        UnicodeRange{ 0x1f442, 0x1f4fc, CodepointWidth::Wide },
-        UnicodeRange{ 0x1f4ff, 0x1f53d, CodepointWidth::Wide },
-        UnicodeRange{ 0x1f54b, 0x1f54e, CodepointWidth::Wide },
-        UnicodeRange{ 0x1f550, 0x1f567, CodepointWidth::Wide },
-        UnicodeRange{ 0x1f57a, 0x1f57a, CodepointWidth::Wide },
-        UnicodeRange{ 0x1f595, 0x1f596, CodepointWidth::Wide },
-        UnicodeRange{ 0x1f5a4, 0x1f5a4, CodepointWidth::Wide },
-        UnicodeRange{ 0x1f5fb, 0x1f64f, CodepointWidth::Wide },
-        UnicodeRange{ 0x1f680, 0x1f6c5, CodepointWidth::Wide },
-        UnicodeRange{ 0x1f6cc, 0x1f6cc, CodepointWidth::Wide },
-        UnicodeRange{ 0x1f6d0, 0x1f6d2, CodepointWidth::Wide },
-        UnicodeRange{ 0x1f6d5, 0x1f6d7, CodepointWidth::Wide },
-        UnicodeRange{ 0x1f6dc, 0x1f6df, CodepointWidth::Wide },
-        UnicodeRange{ 0x1f6eb, 0x1f6ec, CodepointWidth::Wide },
-        UnicodeRange{ 0x1f6f4, 0x1f6fc, CodepointWidth::Wide },
-        UnicodeRange{ 0x1f7e0, 0x1f7eb, CodepointWidth::Wide },
-        UnicodeRange{ 0x1f7f0, 0x1f7f0, CodepointWidth::Wide },
-        UnicodeRange{ 0x1f90c, 0x1f93a, CodepointWidth::Wide },
-        UnicodeRange{ 0x1f93c, 0x1f945, CodepointWidth::Wide },
-        UnicodeRange{ 0x1f947, 0x1f9ff, CodepointWidth::Wide },
-        UnicodeRange{ 0x1fa70, 0x1fa7c, CodepointWidth::Wide },
-        UnicodeRange{ 0x1fa80, 0x1fa88, CodepointWidth::Wide },
-        UnicodeRange{ 0x1fa90, 0x1fabd, CodepointWidth::Wide },
-        UnicodeRange{ 0x1fabf, 0x1fac5, CodepointWidth::Wide },
-        UnicodeRange{ 0x1face, 0x1fadb, CodepointWidth::Wide },
-        UnicodeRange{ 0x1fae0, 0x1fae8, CodepointWidth::Wide },
-        UnicodeRange{ 0x1faf0, 0x1faf8, CodepointWidth::Wide },
-        UnicodeRange{ 0x20000, 0x2fffd, CodepointWidth::Wide },
-        UnicodeRange{ 0x30000, 0x3fffd, CodepointWidth::Wide },
-        UnicodeRange{ 0xe0100, 0xe01ef, CodepointWidth::Ambiguous },
-        UnicodeRange{ 0xf0000, 0xffffd, CodepointWidth::Ambiguous },
-        UnicodeRange{ 0x100000, 0x10fffd, CodepointWidth::Ambiguous },
+    static constexpr std::array<UnicodeRange, 297> s_wideAndAmbiguousTable{

```

Comment:
I've made it so that narrow codepoints aren't stored in this table anymore. This removes 3 entries.